### ISSUE TITLE
Add way to revoke permission

### ISSUE THREAD

**Issue by jadahl** [flatpak/xdg-desktop-portal#235]
The user is supposed to be able to revoke permission of applications. Some preliminary requirements:

* The permission should be revoked immediately - an application that got access to e.g. the microphone revoked shouldn't be able to continue using it.
* The application shouldn't always have to be restarted - force quitting an application might lead to data loss and/or interrupted usage. Might possibly need restarting in certain circumstances (e.g. revoked access to home directory etc) while not in other (revoked access to microphone)
* There needs to be a way for xdg-desktop-portal to signal services outside of the sandbox about permission changes; e.g. a "synchronize permissions" signal that PulseAudio and PipeWire can use to recheck access to devices.

This means that arbitrary components poking directly at the permission store might not work, unless it can be done in a way that makes xdg-desktop-portal perfectly aware of what changed, so the correct signals can be emitted.

**Comment by matthiasclasen** [comment]
yes, we need some change notification from the permission store for this, I think

**Comment by matthiasclasen** [comment]
It turns out the permission store already has a changed signal:

**Comment by matthiasclasen** [comment]
I've added a test for the Changed signal now. Seems to work
